src/Client/Common.php

```diff
@@ -5,18 +5,20 @@
 namespace MatchBot\Client;
 
 use GuzzleHttp\Client;
+use JetBrains\PhpStorm\Pure;
 use Psr\Log\LoggerInterface;
 
 abstract class Common
 {
     private array $clientSettings;
-    private Client $httpClient;
-    protected LoggerInterface $logger;
 
-    public function __construct(array $settings, LoggerInterface $logger)
-    {
+    #[Pure]
+    public function __construct(
+        private array $settings,
+        protected LoggerInterface $logger,
+        private Client $httpClient
```

Comment:
This was not a constructor arg before and I think this actually breaks the client instantiation. Have a look at `Client\Campaign`, Client\Donation` etc. in the dependencies – my IDE highlights them as broken because of the new required constructor arg here.